### ISSUE TITLE
Auto language change according to system settings

### ISSUE THREAD

**Issue by Rakleed** [picotorrent/picotorrent#948]
Please add auto change language according to system settings. And it would be nice to have this option selected by default.

**Comment by vktr** [comment]
This is how it should work already. Are you experiencing another behavior?

Can you give me the steps you follow to reproduce the issue?

**Comment by Rakleed** [comment]
If it already works like that, then great. But for greater clarity, then it would be better to add the item "System" to the list of languages. And so that there are no questions about the fact that if I change the language myself, but then return to the one that was by default, will it correspond to the system language in the future.

**Comment by vktr** [comment]
I agree it would be useful, but I also think it adds complexity for little gain. I think the number of users who regularly changes their OS language is quite small, and the nuisance of having to change the language in PicoTorrent is quite small as well.

**Comment by Rakleed** [comment]
What's the difficulty? It seems to me that users change the system language as rarely as the application language.

**Comment by vktr** [comment]
It is not difficult, but it adds complexity without bringing any real benefits. As you said, users change their system language as rarely as the application langauge - and I agree. The difference is that for those cases where we get the UI language wrong, or don't have the users native language available, they need a way to change the application language. Which we have, in the language drop down.

**Comment by Rakleed** [comment]
So I am not suggesting to completely remove the drop-down list with languages, but to add the "System defaults" item to this list and make this item the default.

**Comment by vktr** [comment]
I understand, and I still think it adds unnecessary complexity. This is why;

We do our best to deduce the user language when we set the language, here https://github.com/picotorrent/picotorrent/blob/master/src/picotorrent/ui/translator.cpp#L131. We try various ways of finding the correct language, defaulting to English if we find none.

When we display the languages list in the preferences dialog, all languages are treated the same. You select an item and that language will be stored in the setting database. Easy. For this use case with a "System default" entry, one if the languages displayed would be different, and we would have to keep track of the difference somehow. Also, there is the case where you pick "System default" and that language isn't available, so we would fall back to English anyway.

I think for 99% of the cases, if PicoTorrent did not pick the correct language when first launching, the user would find it's way to Preferences->Language and change it to whatever language they prefer, and that's it. No more changing languages.

For the 1% that regularly changes their system language *and* want PicoTorrent to match, I think they can go through the effort of heading to Preferences->Languages as well, since a language change in Windows I think would require at least a Logout-Login sequence if not a complete restart.

So for me, it doesn't make sense to support the 1% use case, but I will be more than willing to review a PR adding this if you (or anyone else) find the time to implement it.
